src/Word/DashToSeparator.php

```diff
@@ -14,15 +17,42 @@
  *     ...
  * }
  * @template TOptions of Options
- * @extends AbstractSeparator<TOptions>
+ * @implements FilterInterface<mixed>
  */
-final class DashToSeparator extends AbstractSeparator
+final class DashToSeparator implements FilterInterface
 {
+    protected string $separator = ' ';
+
+    /**
+     * @param Options|string $separator Space by default
+     */
+    public function __construct(string|array $separator = ' ')
```

Comment:
I'm going for consistent constructors throughout, so ideally:

```suggestion
    /** @param Options $separator */
    public function __construct(array $options = [])
```

The `Options` psalm type can also be sealed.

Inside the constructor, I would then apply the separator option to a private readonly property with `$this->separator = $options['separator'] ?? ' ';